torch/cuda/__init__.py

```diff
@@ -317,41 +326,118 @@ def can_device_access_peer(device: _device_t, peer_device: _device_t) -> bool:
         raise AssertionError("Invalid peer device id")
     return torch._C._cuda_canDeviceAccessPeer(device, peer_device)
 
-
-@contextlib.contextmanager
-def stream(stream):
+class StreamContext(object):
     r"""Context-manager that selects a given stream.
-
     All CUDA kernels queued within its context will be enqueued on a selected
     stream.
-
-    Args:
-        stream (Stream): selected stream. This manager is a no-op if it's
+    Arguments:
+        StreamContext (Stream): selected stream. This manager is a no-op if it's
```

Comment:
I don't think you wanted to rename stream to StreamContext here

Reply:
Yes, fixed this now